pkg/nfs/controllerserver.go

```diff
@@ -359,15 +360,16 @@ func (cs *ControllerServer) getVolumeIDFromNfsVol(vol *nfsVolume) string {
 
 // Given a CSI volume id, return a nfsVolume
 func (cs *ControllerServer) getNfsVolFromID(id string) (*nfsVolume, error) {
-	tokens := strings.Split(id, "/")
-	if len(tokens) != totalIDElements {
-		return nil, fmt.Errorf("volume id %q unexpected format: got %v token(s) instead of %v", id, len(tokens), totalIDElements)
+	volRegex := regexp.MustCompile("^([^/]+)/(.*)/([^/]+)$")
```

Comment:
could you also write a ut for this `getNfsVolFromID` function? thanks.

Reply:
I added now some test cases for this function. As I never wrote much in go before so I hope the test is OK. If you think that some changes would be good, let me know and I will adjust the test.